engineio/transport/polling/decoder.go

```diff
@@ -0,0 +1,128 @@
+package polling
+
+import (
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	"github.com/googollee/go-socket.io/engineio/frame"
+)
+
+// decoder decodes frames from the reader.
+type decoder struct {
+	reader    *bufReader
+	lastFrame *frameReader
+}
+
+func newDecoder(buf []byte, r io.Reader) *decoder {
+	return &decoder{
+		reader: newBufReader(buf, r),
+	}
+}
+
+// NextFrame returns a reader to read the next frame.
+func (d *decoder) NextFrame() (frame.Type, io.Reader, error) {
```

Comment:
i suggest return the next type from method:

```go
type FrameReader struct {
	ft frame.Type
	io.Reader
}
```

what do u think about it?

Reply:
Done.